src/renovate.ts

```diff
@@ -0,0 +1,64 @@
+import Docker from './docker';
+import { exec } from '@actions/exec';
+import fs from 'fs';
+import path from 'path';
+
+class Renovate {
+  private configFileEnv = 'RENOVATE_CONFIG_FILE';
+  private tokenEnv = 'RENOVATE_TOKEN';
+  private configFileMountDir = '/github-action';
+
+  private configFile: string;
+  private docker: Docker;
+
+  constructor(configFile: string, private token: string) {
+    this.configFile = path.resolve(configFile);
+
+    this.validateArguments();
+
+    this.docker = new Docker();
+  }
+
+  runDockerContainer(): Promise<void> {
+    return new Promise((resolve, reject) => {
+      const commandArguments = [
+        '--rm',
+        `--env ${this.configFileEnv}=${this.configFileMountPath()}`,
+        `--env ${this.tokenEnv}=${this.token}`,
+        `--volume ${this.configFile}:${this.configFileMountPath()}`,
+        this.docker.image(),
+      ];
+      const command = `docker run ${commandArguments.join(' ')}`;
+
+      exec(command)
+        .then((code) => {
+          if (code === 0) {
+            resolve();
+          } else {
+            reject(new Error(`'docker run' failed with exit code ${code}.`));
+          }
+        })
+        .catch((error) => {
+          reject(error);
+        });
+    });
```

Comment:
```suggestion
    const commandArguments = [
      'run',
      '--rm',
      `--env ${this.configFileEnv}=${this.configFileMountPath()}`,
      `--env ${this.tokenEnv}=${this.token}`,
      `--volume ${this.configFile}:${this.configFileMountPath()}`,
      this.docker.image(),
    ];
    const command = `docker`;
    
    const code = exec(command, commandArguments);
    if (code !== 0) {
      new Error(`'docker run' failed with exit code ${code}.`);
    }
```

Reply:
Got it, thanks for explaining why you prefer async/await :)